File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Location.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "locations", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "locations_unique", columnNames = {"location_name"})
})
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "location_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "street_address_id")
    private StreetAddress streetAddress;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "city_id")
    private City city;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "state_id")
    private State state;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "zip_code_id")
    private ZipCode zipCode;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "country_id")
    private Country country;

    @NotNull
    @Column(name = "location_name", nullable = false, length = Integer.MAX_VALUE)
    private String locationName;

}

File: umbrella-insurance-react-ui/src/models/geographies/locations/v1/Location.ts
import { City } from "../../cities/v1/City";
import { Country } from "../../countries/v1/Country";
import { State } from "../../states/v1/State";
import { StreetAddress } from "../../streetAddresses/v1/StreetAddress";
import { ZipCode } from "../../zipCodes/v1/ZipCode";

export class Location {
    id?: number;
    streetAddress?: StreetAddress;
    city?: City;
    state?: State;
    zipCode?: ZipCode;
    country?: Country;
    locationName?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.locationName) {
                this.locationName = obj.locationName;
            }
            if(obj?.streetAddress) {
                this.streetAddress = new StreetAddress(obj.streetAddress);
            }
            if(obj?.city) {
                this.city = new City(obj.city);
            }
            if(obj?.state) {
                this.state = new State(obj.state);
            }
            if(obj?.zipCode) {
                this.zipCode = new ZipCode(obj.zipCode);
            }
            if(obj?.locationName) {
                this.locationName = obj.locationName;
            }
        }
    }
}